### PR TITLE
Docs: update environment design

### PR DESCRIPTION
**Pull request by hongchaodeng**
Update the design:

- Remove Initializer and replace it with Application + depends-on-app workflow step + env-binding policy + multi-env workflow step.
- Focus on two user scenarios using KubeVela:
  - Using KubeVela to initialize Environment
  - Using KubeVela to compose multi-env application rollout

### CODE REVIEW

**Review by leejanee — APPROVED** [review]
lgtm

**Review by wonderflow — APPROVED** [review]
cool, ping @yangsoon  please review

File: design/vela-core/environment.md [hunk 1]
```diff
@@ -1,294 +1,262 @@
...
+
+  - name: wait-dependencies
+    type: depends-on-app
+    properties:
+      name: ocm-cluster-manager
+      namespace: vela-system
+      
+  - name: apply-self
+    type: apply-application
+```
```
Comment (yangsoon) [inline design/vela-core/environment.md:261]: This demo should be modified
```
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: managed-cluster
  namespace: vela-system
spec:
  components:
    - name: ack-worker
      type: alibaba-ack
      properties:
        writeConnectionSecretToRef:
          name: ack-conn
          namespace: vela-system
  workflow:
    steps:
      - name: wait-dependencies
        type: depends-on-app
        properties:
          name: terraform-alibaba
          namespace: vela-system

      - name: wait-dependencies
        type: depends-on-app
        properties:
          name: ocm-cluster-manager
          namespace: vela-system

      - name: terraform-ack
        type: create-ack
        properties:
          component: ack-worker
        outputs:
          - name: connInfo
            exportKey: connInfo

      - name: register-ack
        type: register-cluster
        inputs:
          - from: connInfo
            parameterKey: connInfo
        properties:
          # user should set public network address of APIServer
          hubAPIServer: {{ public network address of APIServer }}
          env: prod
          initNameSpace: default
          patchLabels:
            purpose: test
```

File: design/vela-core/environment.md [hunk 2]
```diff
@@ -1,294 +1,262 @@
-# Initializing Environments For Application Deployment
+# Making Environments For Application Deployment
 
 ## Background
 
-A team of application development usually needs to initialize some shared environments for developers to deploy applications to.
+A application development team usually needs to initialize some shared environments for developers to deploy applications to.
```
Comment (zzxwill) [inline design/vela-core/environment.md:5]:
```suggestion
An application development team usually needs to initialize some shared environments for developers to deploy applications to.
```

File: design/vela-core/environment.md [hunk 3]
```diff
@@ -1,294 +1,262 @@
...
 and a *prod* environment for running applications to serve live traffic.
-The environment is a logical concept grouping common resources that multiple application deployments would depend on.
 
-A list of resources we want to initialize per environment:
+The environment is a logical concept grouping common resources that many applications would depend on. Below are a list of cases:
 
-- K8s Clusters. Different environments might have clusters of different size and version, e.g.
-  small v1.10 k8s cluster for *dev* environment and large v1.9 k8s cluster for *prod* environment.
-- Admin policies. Production environment would usually set global policies on all application deployments,
+- Kubernetes clusters. These include not only existing clusters but also new clusters to create during environment setup.
```
Comment (zzxwill) [inline design/vela-core/environment.md:11]:
```suggestion
- Kubernetes clusters. These include not only existing clusters but also new clusters to be created during environment setup.
```

File: design/vela-core/environment.md [hunk 4]
```diff
@@ -1,294 +1,262 @@
...
+- Admin policies. Production environment would usually set global policies for all application deployments,
   like chaos testing, SLO requirements, security scanning, misconfiguration detection.
-- Operators & CRDs. Environments contains a variety of Operators & CRDs as system capabilities.
-  These operators can include domain registration, routing, logging, monitoring, autoscaling, etc.
+- System components. That include installing system level Operators & CRDS like FluxCD, KEDA, OpenKruise,
```
Comment (zzxwill) [inline design/vela-core/environment.md:14]:
```suggestion
- System components. That includes installing system level Operators & CRDS like FluxCD, KEDA, OpenKruise,
```

File: design/vela-core/environment.md [hunk 5]
```diff
@@ -1,294 +1,262 @@
...
+  shared Namespaces, Observability (Prometheus, Grafana, Loki), Terraform Controller, KubeFlow Controller, etc.
```
Comment (zzxwill) [inline design/vela-core/environment.md:15]:
```suggestion
  Shared Namespaces, Observability (Prometheus, Grafana, Loki), Terraform Controller, KubeFlow Controller, etc.
```

File: design/vela-core/environment.md [hunk 6]
```diff
@@ -1,294 +1,262 @@
...
-- Platform team initializes Environments using system Components.
-- Platform team defines user Components for Applications.
-- Developers choose user Components and Environments to deploy Applications.
 
-When initializing environments, there are additional requirements:
+## Using KubeVela to initialize Environment
 
-- The ability to describe dependency relations between environments. For example, both *dev* and *prod* environments
-  would depend on a common environment that sets up operators both *dev* and *prod* would use.
+An environment is just a logical concept that groups a bundle of resources for multiple applications to share.
```
Comment (zzxwill) [inline design/vela-core/environment.md:25]: This meaning has been expressed in a line above https://github.com/oam-dev/kubevela/pull/2199/files#diff-b42a63b4159dee9cf6945f592072aa04d2fb83b18a52c85ef3339f9df1f1bf24R9.

File: design/vela-core/environment.md [hunk 7]
```diff
@@ -1,294 +1,262 @@
...
 
-In the following we propose solutions to implement environment concept for application delivery.
+When initializing an environment, there is one additional requirement:
 
-## Initializer CRD
+- The ability to describe dependencies between environment modules. For example, both *dev* and *prod* environments
+  would depend on a common module that installs operators both *dev* and *prod* would use.
 
-We propose to add an Initializer CRD:
+To setup shared resources and define dependencies, we propose to use Application with two Workflow Steps
```
Comment (zzxwill) [inline design/vela-core/environment.md:33]: In the following demo, there is no workflow.

File: design/vela-core/environment.md [hunk 8]
```diff
@@ -1,294 +1,262 @@
...
+to achieve the requirements. An example looks like below:
 
 ```yaml
-kind: Initializer
+apiVersion: core.oam.dev/v1beta1
+kind: Application
```
Comment (zzxwill) [inline design/vela-core/environment.md:38]: I am still curious when an application can be called an environment? I should have missed something important.

File: design/vela-core/environment.md [hunk 9]
```diff
@@ -1,294 +1,262 @@
...
+  - name: <env-policy-name>
+    type: <env-policy-type>
+    properties:
+      <env-policy-props>
 
-## Initializer-Related Operators
+  workflow:
+  - name: wait-dependencies
+    # depends-on-app step will wait until the dependent app's status turns to Running phase
+    type: depends-on-app
```
Comment (zzxwill) [inline design/vela-core/environment.md:59]: Is `workflow` in some application?

For this type, do all components of the application have to be ready first? Why not waiting for a component?

File: design/vela-core/environment.md (hunk 3 above)
Comment (hongchaodeng) [reply design/vela-core/environment.md:11]: 这个原先是正确的语法。。。

File: design/vela-core/environment.md (hunk 5 above)
Comment (hongchaodeng) [reply design/vela-core/environment.md:15]: 这个在 markdown 里没换行呢。。。

File: design/vela-core/environment.md (hunk 8 above)
Comment (hongchaodeng) [reply design/vela-core/environment.md:38]: We will focus on how to use KubeVela to support environment-based application rollout.
We don't call an Application an environment, but focus on how we use KubeVela to support various scenarios.

File: design/vela-core/environment.md (hunk 9 above)
Comment (hongchaodeng) [reply design/vela-core/environment.md:59]: Waiting for an app is good enough for initial use cases.
But definitely we can add more complex operations like waiting for a component.
We can wait until more use cases pops up and implement them later.

File: design/vela-core/environment.md (hunk 6 above)
Comment (hongchaodeng) [reply design/vela-core/environment.md:25]: Well. This sentence is good enough as the beginning of the section.

File: design/vela-core/environment.md (hunk 7 above)
Comment (hongchaodeng) [reply design/vela-core/environment.md:33]: What? It has two steps...

```
  workflow:
  - name: wait-dependencies
    # depends-on-app step will wait until the dependent app's status turns to Running phase
    type: depends-on-app
    properties:
      name: <application name>
      namespace: <application namespace>
      
  - name: apply-self
    # apply-application will apply all components of this Application per se
    type: apply-application
```

File: design/vela-core/environment.md [hunk 10]
```diff
@@ -1,294 +1,262 @@
...
-
-In this case, an Initializer defines shared services to be installed on the cluster.
-Deploying such an initializer will setup these services on the host cluster.
-
-Platform team would define the shared services as Components, and deploy such an environment as an Initializer:
+Then we will create a Kubernetes cluster:
 
 ```yaml
-kind: Initializer
+apiVersion: core.oam.dev/v1beta1
```
Comment (yangsoon) [inline design/vela-core/environment.md:210]: this demo should be fixed

```yaml
apiVersion: core.oam.dev/v1beta1
kind: Application
metadata:
  name: managed-cluster
  namespace: vela-system
spec:
  components:
    - name: ack-worker
      type: alibaba-ack
      properties:
        writeConnectionSecretToRef:
          name: ack-conn
          namespace: vela-system
  workflow:
    steps:
      - name: wait-dependencies
        type: depends-on-app
        properties:
          name: terraform-alibaba
          namespace: vela-system

      - name: wait-dependencies
        type: depends-on-app
        properties:
          name: ocm-cluster-manager
          namespace: vela-system
          
      - name: terraform-ack
        type: create-ack
        properties:
          component: ack-worker
        outputs:
          - name: connInfo
            exportKey: connInfo

      - name: register-ack
        type: register-cluster
        inputs:
          - from: connInfo
            parameterKey: connInfo
        properties:
          # user should set public network address of APIServer
          hubAPIServer: {{ public network address of APIServer }}
          env: prod
          initNameSpace: default
          patchLabels:
            purpose: test
```

File: design/vela-core/environment.md (hunk 5 above)
Comment (zzxwill) [reply design/vela-core/environment.md:15]: Sorry.

File: design/vela-core/environment.md (hunk 10 above)
Comment (zzxwill) [reply design/vela-core/environment.md:210]: Should the second step name `wait-dependencies` be renamed?

**Review by zzxwill — APPROVED** [review]
/lgtm

**Pull request merged**